File: clients/csharp/Model/Player.cs
namespace AiCup2019.Model
{
    public struct Player
    {
        public int Id { get; set; }
        public int Score { get; set; }
        public Player(int id, int score)
        {
            this.Id = id;
            this.Score = score;
        }
        public static Player ReadFrom(System.IO.BinaryReader reader)
        {
            var result = new Player();
            result.Id = reader.ReadInt32();
            result.Score = reader.ReadInt32();
            return result;
        }
        public void WriteTo(System.IO.BinaryWriter writer)
        {
            writer.Write(Id);
            writer.Write(Score);
        }
    }
}

File: clients/fsharp/Model/Player.fs
#nowarn "0058"
namespace AiCup2019.Model
type Player = {
    Id: int;
    Score: int;
    } with
    member this.writeTo(writer: System.IO.BinaryWriter) =
        writer.Write this.Id
        writer.Write this.Score
    static member readFrom(reader: System.IO.BinaryReader) = {
        Id = reader.ReadInt32()
        Score = reader.ReadInt32()
    }
